spatialmath/box.go

```diff
@@ -352,3 +352,102 @@ func separatingAxisTest(positionDelta, plane r3.Vector, halfSizeA, halfSizeB [3]
 	}
 	return sum
 }
+
+// ToPointCloud converts a box geometry into a []r3.Vector.
+func (b *box) ToPointCloud(options map[string]interface{}) []r3.Vector {
+	// check for user defined spacing
+	var faces []r3.Vector
+	var iter float64
+	if options["resolution"] != nil {
+		iter = options["resolution"].(float64)
+	} else {
+		iter = 0.15 // default spacing
+	}
+	// TODO: the for loops below can be made concurrent if the ToPointCloud method is too slow
+	// create points on box faces with box centered at (0, 0, 0)
+	for i := 0.0; i <= b.halfSize[0]; i += iter {
```

Comment:
These three nearly-identical blocks look really ripe for being abstracted into a function

Reply:
Was going to say the same thing.  Something like this might work
```
	sampleFaces := func(fixedDimension int) {
		starts := [3]float64{0.0, 0.0, 0.0}
		starts[fixedDimension] = b.halfSize[fixedDimension]
		for i := starts[0]; i <= b.halfSize[0]; i += iter {
			for j := starts[1]; i <= b.halfSize[1]; i += iter {
				for k := starts[2]; k <= b.halfSize[2]; k += iter {
					// front and back faces
					p1 := r3.Vector{i, j, k}
					p2 := r3.Vector{i, j, -k}
					p3 := r3.Vector{-i, j, k}
					p4 := r3.Vector{-i, j, -k}
					p5 := r3.Vector{-i, -j, -k}
					p6 := r3.Vector{-i, -j, k}
					p7 := r3.Vector{i, -j, -k}
					p8 := r3.Vector{i, -j, k}

					switch {
					case i == 0.0 && j == 0.0:
						faces = append(faces, p1, p2)
					case j == 0.0 && k == 0.0:
						faces = append(faces, p2, p3)
					case i == 0.0 && k == 0.0:
						faces = append(faces, p4, p5)
					case i == 0.0:
						faces = append(faces, p1, p2, p5, p6)
					case j == 0.0:
						faces = append(faces, p1, p2, p3, p4)
					case k == 0.0:
						faces = append(faces, p1, p3, p6, p8)
					default:
						faces = append(faces, p1, p2, p3, p4, p5, p6, p7, p8)
					}
				}
			}
		}
	}

	for i := 0; i < 3; i++ {
		sampleFaces(i)
	}
```

Reply:
still need todo. will push code through for next round of comments